packages/frontend/src/contexts/unirep.js

```diff
@@ -31,7 +31,15 @@ export default class Unirep {
 
   async searchForId(id) {
     const url = new URL(`api/unirep/search/${id}`, SERVER)
-    const type = await fetch(url.toString()).then((r) => r.json())
+    const { type, data } = await fetch(url.toString()).then((r) => r.json())
+    if (type === 'user') {
+      this.signUpsByUserId.set(data[0].commitment, data)
+    } else if (type === 'epochKey') {
+      this.attestationsByEpochKey.set(data[0].epochKey, data)
+    } else if (type === 'attester') {
+      this.deploymentsById.set(data._id, data)
+    }
+    console.log(type, data)
     return type
   }
```

Comment:
loading the data requested by search() into the mappings that will be accessed on the new page; now not calling the functions that would load these on `AttesterPage` `UserPage` and `EpochKeyPage`.

Reply:
I think you should call those functions on an as-needed basis. It's possible that the user navigates directly to those pages (e.g. from clicking a link on another website). In that case the page does need to load the data.